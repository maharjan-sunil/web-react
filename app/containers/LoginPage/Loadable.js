import React from 'react';
import loadable from 'utils/loadable';
import LoaderIndicator from 'components/Loader';

export const LoginContainer = loadable(() => import('./index'), {
  fallback: <LoaderIndicator />,
});

export const ActivateContainer = loadable(() => import('./activate'), {
  fallback: <LoaderIndicator />,
});

export const ForgotPasswordContainer = loadable(
  () => import('./forgotPassword'),
  {
    fallback: <LoaderIndicator />,
  },
);
