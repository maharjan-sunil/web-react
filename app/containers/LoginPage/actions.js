/*
 *
 * AccountPage actions
 *
 */

import {
  LOGIN_USER_SUCCESS,
  LOGIN_USER_ERROR,
  LOGIN_USER,
  ACTIVATE_USER,
  RESEND_ACTIVATION_CODE_USER,
  RESEND_ACTIVATION_CODE_USER_SUCCESS,
  RESET_STATUSCODE_ACTION,
  ACTIVATION_USER_ERROR,
  RESET_STATE_ACTION,
  EMAIL_VERIFY_ACTION,
  UPDATE_PASSWORD_ACTION,
  FORGOT_PASSWORD_SUCCESS,
  FORGOT_PASSWORD_ERROR,
} from './constants';

export function login(email, password) {
  return {
    type: LOGIN_USER,
    email,
    password,
    dateTimeOffSet: new Date().getTimezoneOffset(),
  };
}

export function loginUserSuccess(request, response) {
  return {
    type: LOGIN_USER_SUCCESS,
    response,
    request,
  };
}

export function loginUserError(error) {
  return {
    type: LOGIN_USER_ERROR,
    error,
  };
}

export function activateUserAction(
  email,
  newPassword,
  confirmPassword,
  activationCode,
) {
  return {
    type: ACTIVATE_USER,
    email,
    newPassword,
    confirmPassword,
    activationCode,
    dateTimeOffSet: new Date().getTimezoneOffset(),
  };
}

export function activationUserError(error) {
  return {
    type: ACTIVATION_USER_ERROR,
    error,
  };
}
export function resendActivationCodeAction(email) {
  return {
    type: RESEND_ACTIVATION_CODE_USER,
    email,
  };
}

export function resendActivationCodeSuccess(response) {
  return {
    type: RESEND_ACTIVATION_CODE_USER_SUCCESS,
    response,
  };
}

export function resetStatusCodeAction() {
  return {
    type: RESET_STATUSCODE_ACTION,
  };
}

export function resetStateAction() {
  return {
    type: RESET_STATE_ACTION,
  };
}

export function emailVerifyAction(email) {
  return {
    type: EMAIL_VERIFY_ACTION,
    email,
    dateTimeOffSet: new Date().getTimezoneOffset(),
  };
}
export function updatePasswordAction(
  email,
  code,
  newPassword,
  confirmPassword,
) {
  return {
    type: UPDATE_PASSWORD_ACTION,
    email,
    newPassword,
    confirmPassword,
    code,
    dateTimeOffSet: new Date().getTimezoneOffset(),
  };
}

export function forgotPasswordSuccess(isVerifyEmail, isVerifyCode, response) {
  return {
    type: FORGOT_PASSWORD_SUCCESS,
    response,
    isVerifyEmail,
    isVerifyCode,
  };
}

export function forgotPasswordError(isVerifyEmail, isVerifyCode, response) {
  return {
    type: FORGOT_PASSWORD_ERROR,
    response,
    isVerifyEmail,
    isVerifyCode,
  };
}
