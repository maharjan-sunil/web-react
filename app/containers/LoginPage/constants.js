/*
 *
 * AccountPage constants
 *
 */

export const LOGIN_USER = 'LoginUser';
export const LOGIN_USER_SUCCESS = 'LoginUserSuccess';
export const LOGIN_USER_ERROR = 'LoginUserError';
export const ACTIVATE_USER = 'ActivateUser';
export const ACTIVATE_USER_SUCCESS = 'ActivateUserSuccess';
export const RESEND_ACTIVATION_CODE_USER = 'ResendActivateCodeUser';
export const RESEND_ACTIVATION_CODE_USER_SUCCESS =
  'ResendActivateCodeUserSuccess';
export const RESET_STATUSCODE_ACTION = 'ResetStatusCodeUser';
export const ACTIVATION_USER_ERROR = 'ActivateUserError';
export const RESET_STATE_ACTION = 'ResetStateUser';
export const EMAIL_VERIFY_ACTION = 'VerifyEmailUser';
export const UPDATE_PASSWORD_ACTION = 'UpdatePasswordUser';
export const FORGOT_PASSWORD_SUCCESS = 'ForgotPasswordSuccess';
export const FORGOT_PASSWORD_ERROR = 'ForgotPasswordError';
