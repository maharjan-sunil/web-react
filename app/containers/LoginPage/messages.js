/*
 * LoginPage Messages
 *
 * This contains all the text for the LoginPage component.
 */
import { defineMessages } from 'react-intl';

export const scope = 'boilerplate.containers.LoginPage';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is Login Page.',
  },
  invalidLogin: {
    id: `${scope}.invalid_Login`,
    defaultMessage: 'Invalid login credentials ',
  },
  activationFailed: {
    id: `${scope}.activation_Failed`,
    defaultMessage: 'Activation Failed',
  },
  activationSuccess: {
    id: `${scope}.activation_Success`,
    defaultMessage: 'Activation code successfully send.',
  },
  emailVerified: {
    id: `${scope}.email_Verified`,
    defaultMessage: 'Email Verified and code send to Email.',
  },
  emailVerifiedFailed: {
    id: `${scope}.email_Verified_Failed`,
    defaultMessage: 'Email not exist, Please use correct email.',
  },
  codeVerifiedFailed: {
    id: `${scope}.code_Verified_Failed`,
    defaultMessage: 'Code doesnot match, Please use correct code.',
  },
});
