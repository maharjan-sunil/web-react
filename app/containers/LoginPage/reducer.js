import produce from 'immer';
import {
  LOGIN_USER,
  LOGIN_USER_SUCCESS,
  LOGIN_USER_ERROR,
  RESET_STATUSCODE_ACTION,
  RESEND_ACTIVATION_CODE_USER,
  RESEND_ACTIVATION_CODE_USER_SUCCESS,
  ACTIVATION_USER_ERROR,
  RESET_STATE_ACTION,
  FORGOT_PASSWORD_SUCCESS,
  FORGOT_PASSWORD_ERROR,
  EMAIL_VERIFY_ACTION,
  UPDATE_PASSWORD_ACTION,
  ACTIVATE_USER,
} from './constants';

export const initialState = {
  loginLoader: false,
  buttonLoader: false,
  loginSuccess: false,
  response: false,
  statusCode: 0,
  activationUser: false,
  isVerifyEmail: false,
  isVerifyCode: false,
  error: false,
};

/* eslint-disable default-case, no-param-reassign */
const loginPageReducer = (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case LOGIN_USER:
        draft.loginLoader = true;
        break;
      case EMAIL_VERIFY_ACTION:
        draft.loginLoader = true;
        break;
      case UPDATE_PASSWORD_ACTION:
        draft.loginLoader = true;
        break;
      case ACTIVATE_USER:
        draft.loginLoader = true;
        break;
      case LOGIN_USER_SUCCESS:
        // add to global State CurrentUser
        draft.response = { ...state.response, ...action };
        draft.loginLoader = false;
        if (action.response.returnUrl === `${process.env.API}/activate`) {
          draft.loginSuccess = false;
          draft.statusCode = 403;
          localStorage.setItem('activateEmail', action.request.email);
          draft.activationUser = true;
        } else {
          draft.loginSuccess = true;
          draft.activationUser = false;
        }
        break;
      case LOGIN_USER_ERROR:
        draft.response = { ...state.response, ...action };
        draft.loginSuccess = false;
        draft.activationUser = false;
        draft.loginLoader = false;
        break;
      case ACTIVATION_USER_ERROR:
        draft.response = { ...state.response, ...action };
        draft.activationUser = true;
        draft.loginSuccess = false;
        draft.loginLoader = false;
        break;
      case RESEND_ACTIVATION_CODE_USER:
        draft.buttonLoader = true;
        break;
      case RESEND_ACTIVATION_CODE_USER_SUCCESS:
        draft.buttonLoader = false;
        draft.response = action.response;
        draft.statusCode = 0;
        break;
      case RESET_STATUSCODE_ACTION:
        draft.statusCode = 0;
        if (state.response.error) {
          state.response.error = false;
        }
        draft.response = { ...state.response, ...action };
        break;
      case RESET_STATE_ACTION:
        if (state.response.error) {
          state.response.error = false;
        }
        draft.response = { ...state.response, ...action };
        draft.error = false;
        draft.isVerifyEmail = false;
        draft.isVerifyCode = false;
        draft.loginSuccess = false;
        break;
      case FORGOT_PASSWORD_SUCCESS:
        draft.loginLoader = false;
        draft.response = action.response;
        draft.isVerifyEmail = action.isVerifyEmail;
        draft.isVerifyCode = action.isVerifyCode;
        draft.error = false;
        draft.loginSuccess = false;
        break;
      case FORGOT_PASSWORD_ERROR:
        draft.loginLoader = false;
        draft.response = { ...state.response, ...action };
        draft.isVerifyEmail = action.isVerifyEmail;
        draft.isVerifyCode = action.isVerifyCode;
        draft.error = true;
        draft.loginSuccess = false;
        break;
    }
  });

export default loginPageReducer;
