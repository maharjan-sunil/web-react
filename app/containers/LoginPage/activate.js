/* eslint-disable indent */
import React, { memo, useState } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';
import { useForm } from 'react-hook-form';
import PropTypes from 'prop-types';
import { Link, Redirect } from 'react-router-dom';
import { Key } from '@styled-icons/fa-solid/Key';
import { useInjectSaga } from 'utils/injectSaga';
import { useInjectReducer } from 'utils/injectReducer';
import { Form, Label, Button, Img, Alert } from '@bootstrap-styled/v4';
import LinkButton from 'components/LinkButton';
import auth from 'helpers/auth';
import { LoginLoader, ButtonLoader } from 'components/Loader';
import reducer from './reducer';
import saga from './saga';
import messages from './messages';
import { makeSelectResponse } from './selectors';
import {
  LoginBody,
  LoginBox,
  LoginWrap,
  Logo,
  SubApp,
  StyledFormGroup,
  StyledInput,
  StyledInputGroupWrap,
} from './styles';

import WallbeeLogo from './WallbeeLogo.png';
import {
  activateUserAction,
  resendActivationCodeAction,
  resetStatusCodeAction,
} from './actions';

export function Activate({
  location,
  onSubmitForm,
  state,
  onResendActivation,
  on403,
}) {
  useInjectReducer({ key: 'loginPage', reducer });
  useInjectSaga({ key: 'loginPage', saga });

  const { register, handleSubmit, errors, setValue, watch } = useForm();
  const [{ email }, setDefaultValue] = useState({
    email: '',
  });
  React.useEffect(() => {
    register({ name: 'email' }, { required: true });
    register({ name: 'newPassword' }, { required: true });
    register(
      { name: 'confirmPassword' },
      { required: true, validate: value => value === watch('newPassword') },
    );
    register({ name: 'activationCode' }, { required: true });
    if (localStorage.getItem('activateEmail') !== '') {
      setDefaultValue({
        email: localStorage.getItem('activateEmail'),
      });
      setValue('email', localStorage.getItem('activateEmail'));
    }
    on403();
  }, []);

  let action;
  if (state.response.error && !state.loginSuccess) {
    action = (
      <Alert color="danger">{messages.activationFailed.defaultMessage}</Alert>
    );
  } else if (state.response === true) {
    action = (
      <Alert color="success">{messages.activationSuccess.defaultMessage}</Alert>
    );
  } else if (state.loginSuccess && auth.isAuthenticated()) {
    const redirectTo = location.state !== undefined ? location.state.from : '/';
    action = <Redirect to={redirectTo} />;
  }

  return (
    <React.Fragment>
      <LoginBody />
      <LoginBox>
        <Form onSubmit={handleSubmit(onSubmitForm)}>
          <LoginWrap>
            <Logo>
              <Img alt="wallbee-logo" src={WallbeeLogo} />
            </Logo>
            <SubApp>Control System</SubApp>
          </LoginWrap>
          {action}
          <StyledFormGroup>
            <StyledInput
              placeholder="Email"
              name="email"
              readOnly="readOnly"
              autoFocus
              defaultValue={email}
              onChange={e => {
                setValue('email', e.currentTarget.value, {
                  shouldValidate: true,
                  shouldDirty: true,
                });
              }}
            />
            <Label htmlFor="email">Email</Label>
          </StyledFormGroup>
          {errors.email && <p className="text-danger">Email is required</p>}
          <StyledFormGroup>
            <StyledInput
              type="password"
              placeholder="New Password"
              name="newPassword"
              onChange={e => {
                setValue('newPassword', e.currentTarget.value, {
                  shouldValidate: true,
                  shouldDirty: true,
                });
              }}
            />
            <Label htmlFor="newPassword">New Password</Label>
          </StyledFormGroup>
          {errors.newPassword && errors.newPassword.type === 'required' && (
            <p className="text-danger">New Password is required</p>
          )}
          <StyledFormGroup>
            <StyledInput
              type="password"
              placeholder="Confirm Password"
              name="confirmPassword"
              onChange={e => {
                setValue('confirmPassword', e.currentTarget.value, {
                  shouldValidate: true,
                  shouldDirty: true,
                });
              }}
            />
            <Label htmlFor="confirmPassword">Confirm Password</Label>
          </StyledFormGroup>
          {errors.confirmPassword &&
            errors.confirmPassword.type === 'required' && (
              <p className="text-danger">Confirm Password is required</p>
            )}
          {errors.confirmPassword &&
            errors.confirmPassword.type === 'validate' && (
              <p className="text-danger">Confirm Password not match</p>
            )}
          <StyledInputGroupWrap>
            <div>
              <StyledFormGroup>
                <StyledInput
                  type="text"
                  placeholder="Activation Code"
                  name="activationCode"
                  onChange={e => {
                    setValue('activationCode', e.currentTarget.value, {
                      shouldValidate: true,
                      shouldDirty: true,
                    });
                  }}
                />
                <Label htmlFor="activationCode">Activation Code</Label>
              </StyledFormGroup>
              {errors.activationCode && (
                <p className="text-danger">Activation Code is required</p>
              )}
            </div>
            <div>
              <Button
                data-toggle="tooltip"
                data-placement="top"
                title="Resend activation code"
                className="btn"
                disabled={state.buttonLoader}
                onClick={() => {
                  onResendActivation(email);
                }}
              >
                {state.buttonLoader ? <ButtonLoader /> : <Key size="12" />}
              </Button>
            </div>
          </StyledInputGroupWrap>
          <Button
            color="primary"
            size="md"
            type="submit"
            className="mr-1"
            disabled={state.loginLoader}
          >
            Activate
          </Button>
          <LinkButton color="link" size="md" tag={Link} to="/login">
            Sign in with another user
          </LinkButton>
          {state.loginLoader && <LoginLoader message="Activating User" />}
          <p className="mt-5 mb-3 text-muted text-center">
            &copy; WallBee 2020
          </p>
        </Form>
      </LoginBox>
    </React.Fragment>
  );
}

Activate.propTypes = {
  onSubmitForm: PropTypes.func,
  onResendActivation: PropTypes.func,
  state: PropTypes.object,
  location: PropTypes.object,
  on403: PropTypes.func,
};

const mapStateToProps = createStructuredSelector({
  state: makeSelectResponse(),
});

export function mapDispatchToProps(dispatch) {
  return {
    onSubmitForm: data => {
      dispatch(
        activateUserAction(
          data.email,
          data.newPassword,
          data.confirmPassword,
          data.activationCode,
        ),
      );
    },
    onResendActivation: email => {
      dispatch(resendActivationCodeAction(email));
    },
    on403: () => {
      dispatch(resetStatusCodeAction());
    },
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(
  withConnect,
  memo,
)(Activate);
