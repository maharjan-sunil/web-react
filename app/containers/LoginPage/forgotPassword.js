/* eslint-disable no-unused-vars */
/* eslint-disable no-nested-ternary */
/* eslint-disable indent */
import React, { memo, useState } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';
import { useForm } from 'react-hook-form';
import PropTypes from 'prop-types';
import { Link, Redirect } from 'react-router-dom';
import { useInjectSaga } from 'utils/injectSaga';
import { useInjectReducer } from 'utils/injectReducer';
import { Form, Label, Button, Img, Alert } from '@bootstrap-styled/v4';
import LinkButton from 'components/LinkButton';
import { Key } from '@styled-icons/fa-solid/Key';
import { LoginLoader, ButtonLoader } from 'components/Loader';
import reducer from './reducer';
import saga from './saga';
import messages from './messages';
import { makeSelectResponse } from './selectors';
import {
  LoginBody,
  LoginBox,
  LoginWrap,
  Logo,
  SubApp,
  StyledFormGroup,
  StyledInput,
  StyledInputGroupWrap,
} from './styles';
import WallbeeLogo from './WallbeeLogo.png';
import {
  emailVerifyAction,
  updatePasswordAction,
  resetStateAction,
  resendActivationCodeAction,
} from './actions';

export function ForgotPassword({
  location,
  onSubmitForm,
  state,
  onLoadResetState,
  onEmailVerify,
  onResendActivation,
}) {
  useInjectReducer({ key: 'loginPage', reducer });
  useInjectSaga({ key: 'loginPage', saga });

  const { register, handleSubmit, errors, setValue, watch } = useForm();
  const [{ isEmailVerified }, setDefaultValue] = useState({
    isEmailVerified: false,
  });
  const [{ email }, setEmail] = useState({
    email: '',
  });

  if (state.response === true && !isEmailVerified) {
    setDefaultValue({
      isEmailVerified: true,
    });
    register({ name: 'code' }, { required: true });
    register({ name: 'newPassword' }, { required: true });
    register(
      { name: 'confirmPassword' },
      { required: true, validate: value => value === watch('newPassword') },
    );
  }
  React.useEffect(() => {
    register({ name: 'email' }, { required: true });
    onLoadResetState();
  }, []);

  let action;
  if (state.error && !state.isVerifyEmail) {
    action = (
      <Alert color="danger">
        {messages.emailVerifiedFailed.defaultMessage}
      </Alert>
    );
  } else if (state.error && state.isVerifyEmail && !state.isVerifyCode) {
    action = (
      <Alert color="danger">{messages.codeVerifiedFailed.defaultMessage}</Alert>
    );
  } else if (state.response && state.isVerifyEmail && !state.isVerifyCode) {
    action = (
      <Alert color="success">{messages.emailVerified.defaultMessage}</Alert>
    );
  } else if (state.response && state.isVerifyEmail && state.isVerifyCode) {
    const redirectTo = location.state !== undefined ? location.state.from : '/';
    action = <Redirect to={redirectTo} />;
  }

  return (
    <React.Fragment>
      <LoginBody />
      <LoginBox>
        <Form
          onSubmit={handleSubmit(
            !isEmailVerified ? onEmailVerify : onSubmitForm,
          )}
        >
          <LoginWrap>
            <Logo>
              <Img alt="wallbee-logo" src={WallbeeLogo} />
            </Logo>
            <SubApp>Control System</SubApp>
          </LoginWrap>
          {action}
          <StyledFormGroup>
            <StyledInput
              placeholder="Email"
              name="email"
              readOnly={!!isEmailVerified}
              autoFocus
              onChange={e => {
                setValue('email', e.currentTarget.value, {
                  shouldValidate: true,
                  shouldDirty: true,
                });
                setEmail({
                  email: e.currentTarget.value,
                });
              }}
            />
            <Label htmlFor="email">Email</Label>
          </StyledFormGroup>
          {errors.email && <p className="text-danger">Email is required</p>}
          {isEmailVerified && (
            <StyledInputGroupWrap>
              <div>
                <StyledFormGroup>
                  <StyledInput
                    type="text"
                    placeholder="code"
                    name="code"
                    onChange={e => {
                      setValue('code', e.currentTarget.value, {
                        shouldValidate: true,
                        shouldDirty: true,
                      });
                    }}
                  />
                  <Label htmlFor="code">Code</Label>
                </StyledFormGroup>
              </div>
              <div>
                <Button
                  data-toggle="tooltip"
                  data-placement="top"
                  title="Resend activation code"
                  className="btn"
                  disabled={state.buttonLoader}
                  onClick={() => {
                    onResendActivation(email);
                  }}
                >
                  {state.buttonLoader ? <ButtonLoader /> : <Key size="12" />}
                </Button>
              </div>
            </StyledInputGroupWrap>
          )}

          {isEmailVerified && errors.code && (
            <p className="text-danger">Code is required</p>
          )}
          {isEmailVerified && (
            <StyledFormGroup>
              <StyledInput
                type="password"
                placeholder="New Password"
                name="newPassword"
                onChange={e => {
                  setValue('newPassword', e.currentTarget.value, {
                    shouldValidate: true,
                    shouldDirty: true,
                  });
                }}
              />
              <Label htmlFor="newPassword">New Password</Label>
            </StyledFormGroup>
          )}

          {isEmailVerified &&
            errors.newPassword &&
            errors.newPassword.type === 'required' && (
              <p className="text-danger">New Password is required</p>
            )}
          {isEmailVerified && (
            <StyledFormGroup>
              <StyledInput
                type="password"
                placeholder="Confirm Password"
                name="confirmPassword"
                onChange={e => {
                  setValue('confirmPassword', e.currentTarget.value, {
                    shouldValidate: true,
                    shouldDirty: true,
                  });
                }}
              />
              <Label htmlFor="confirmPassword">Confirm Password</Label>
            </StyledFormGroup>
          )}

          {isEmailVerified &&
            errors.confirmPassword &&
            errors.confirmPassword.type === 'required' && (
              <p className="text-danger">Confirm Password is required</p>
            )}
          {isEmailVerified &&
            errors.confirmPassword &&
            errors.confirmPassword.type === 'validate' && (
              <p className="text-danger">Confirm Password not match</p>
            )}
          <Button
            color="primary"
            size="md"
            type="submit"
            className="mr-1"
            disabled={state.loginLoader}
          >
            Submit
          </Button>
          <LinkButton color="link" size="md" tag={Link} to="/login">
            Sign in with another user
          </LinkButton>
          {state.loginLoader && isEmailVerified && (
            <LoginLoader message="Authenticating User" />
          )}
          {state.loginLoader && !isEmailVerified && (
            <LoginLoader message="Verifying Email" />
          )}
          <p className="mt-5 mb-3 text-muted text-center">
            &copy; WallBee 2020
          </p>
        </Form>
      </LoginBox>
    </React.Fragment>
  );
}

ForgotPassword.propTypes = {
  onSubmitForm: PropTypes.func,
  state: PropTypes.object,
  location: PropTypes.object,
  onLoadResetState: PropTypes.func,
  onEmailVerify: PropTypes.func,
  onResendActivation: PropTypes.func,
};

const mapStateToProps = createStructuredSelector({
  state: makeSelectResponse(),
});

export function mapDispatchToProps(dispatch) {
  return {
    onEmailVerify: data => {
      dispatch(emailVerifyAction(data.email));
    },
    onSubmitForm: data => {
      dispatch(
        updatePasswordAction(
          data.email,
          data.code,
          data.newPassword,
          data.confirmPassword,
        ),
      );
    },
    onLoadResetState: () => {
      dispatch(resetStateAction());
    },
    onResendActivation: email => {
      dispatch(resendActivationCodeAction(email));
    },
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(
  withConnect,
  memo,
)(ForgotPassword);
