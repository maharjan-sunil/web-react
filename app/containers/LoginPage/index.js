/* eslint-disable jsx-a11y/alt-text */
import React, { memo } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';
import { useForm } from 'react-hook-form';
import PropTypes from 'prop-types';
import { Link, Redirect } from 'react-router-dom';

import { useInjectSaga } from 'utils/injectSaga';
import { useInjectReducer } from 'utils/injectReducer';
import { Form, Label, Button, Img, Alert } from '@bootstrap-styled/v4';
import auth from 'helpers/auth';
import LinkButton from 'components/LinkButton';
import { LoginLoader } from 'components/Loader';
import reducer from './reducer';
import saga from './saga';
import messages from './messages';
import { makeSelectResponse } from './selectors';
import {
  LoginBody,
  LoginBox,
  LoginWrap,
  Logo,
  SubApp,
  StyledFormGroup,
  StyledInput,
} from './styles';
import WallbeeLogo from './WallbeeLogo.png';
import { login } from './actions';

export function LoginPage({ location, onSubmitForm, state }) {
  useInjectReducer({ key: 'loginPage', reducer });
  useInjectSaga({ key: 'loginPage', saga });

  const { register, handleSubmit, errors, setValue } = useForm();

  React.useEffect(() => {
    register({ name: 'email' }, { required: true });
    register({ name: 'password' }, { required: true });
    localStorage.removeItem('activateEmail');
    localStorage.removeItem('token');
    localStorage.removeItem('expiresAt');
  }, []);

  let action;
  if (!state.loginSuccess && state.statusCode === 403) {
    action = <Redirect to="/activate" />;
  } else if (
    state.response.error &&
    !state.loginSuccess &&
    !state.activationUser
  ) {
    action = (
      <Alert color="danger">{messages.invalidLogin.defaultMessage}</Alert>
    );
  } else if (state.loginSuccess && auth.isAuthenticated()) {
    const redirectTo = location.state !== undefined ? location.state.from : '/';
    action = <Redirect to={redirectTo} />;
  }

  return (
    <React.Fragment>
      <LoginBody />
      <LoginBox>
        <Form onSubmit={handleSubmit(onSubmitForm)}>
          <LoginWrap>
            <Logo>
              <Img alt="wallbee-logo" src={WallbeeLogo} />
            </Logo>
            <SubApp>Control System</SubApp>
          </LoginWrap>
          {action}
          <StyledFormGroup>
            <StyledInput
              placeholder="Email"
              name="email"
              autoFocus
              onChange={e => {
                setValue('email', e.currentTarget.value, {
                  shouldValidate: true,
                  shouldDirty: true,
                });
              }}
            />
            <Label htmlFor="email">Email</Label>
          </StyledFormGroup>
          {errors.email && <p className="text-danger">Email is required</p>}
          <StyledFormGroup>
            <StyledInput
              type="password"
              placeholder="Password"
              name="password"
              onChange={e => {
                setValue('password', e.currentTarget.value, {
                  shouldValidate: true,
                  shouldDirty: true,
                });
              }}
            />
            <Label htmlFor="password">Password</Label>
          </StyledFormGroup>
          {errors.password && (
            <p className="text-danger">Password is required</p>
          )}
          <Button
            color="primary"
            size="md"
            type="submit"
            className="mr-1"
            disabled={state.loginLoader}
          >
            Log In
          </Button>
          <LinkButton color="link" size="md" tag={Link} to="/forgot-password">
            Forgot Password ?
          </LinkButton>
          {state.loginLoader && <LoginLoader message="Authenticating User" />}
          <p className="mt-5 mb-3 text-muted text-center">
            &copy; WallBee 2020
          </p>
        </Form>
      </LoginBox>
    </React.Fragment>
  );
}

LoginPage.propTypes = {
  onSubmitForm: PropTypes.func,
  state: PropTypes.object,
  location: PropTypes.object,
};

const mapStateToProps = createStructuredSelector({
  state: makeSelectResponse(),
});

export function mapDispatchToProps(dispatch) {
  return {
    onSubmitForm: data => dispatch(login(data.email, data.password)),
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(
  withConnect,
  memo,
)(LoginPage);
