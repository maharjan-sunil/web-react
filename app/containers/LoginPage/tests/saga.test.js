/**
 * Tests for HomePage sagas
 */

import { takeLatest, put } from 'redux-saga/effects';

import { LOGIN_USER } from 'containers/LoginPage/constants';
import { loginUserSuccess, loginUserError } from 'containers/LoginPage/actions';

import loginValidate, { loginPageSaga } from '../saga';

/* eslint-disable redux-saga/yield-effects */
describe('loginPageSaga Saga', () => {
  let loginGenerator;
  const action = {
    email: '',
    password: '',
    type: '',
    dateTimeOffSet: '',
  };

  beforeEach(() => {
    loginGenerator = loginPageSaga(action);

    const selectDescriptor = loginGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('should call the loginUserSuccess action if the response success', () => {
    const response = {
      response: {},
    };
    const putDescriptor = loginGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(loginUserSuccess(response)));
  });

  it('should call the loginUserError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = loginGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(loginUserError(response)));
  });
});

describe('loginValidate Saga', () => {
  const loginValidateSaga = loginValidate();

  it('should start task to watch for LOGIN_USER action', () => {
    const takeLatestDescriptor = loginValidateSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(LOGIN_USER, loginPageSaga));
  });
});
