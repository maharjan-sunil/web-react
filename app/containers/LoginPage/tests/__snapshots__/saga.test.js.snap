// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loginPageSaga Saga should call the loginUserError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/login",
      Object {
        "dateTimeOffSet": "",
        "email": "",
        "password": "",
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`loginPageSaga Saga should call the loginUserSuccess action if the response success 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/login",
      Object {
        "dateTimeOffSet": "",
        "email": "",
        "password": "",
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
